### PR TITLE
Barlow twins for Contrastive SSL migrate to keras 3

### PR DESCRIPTION
**Pull request by maitry63**
This PR Migrate the Barlow twins for Contrastive SSL tutorial into Keras 3.

Colab file: [Notebook ](https://colab.sandbox.google.com/gist/maitry63/f5df8cb3d212fa79b0d55c79ac9b8c2a/copy-of-barlow_twins.ipynb#scrollTo=yw7sX7rlvb6F)

### CODE REVIEW

File: examples/vision/barlow_twins.py [hunk 1]
```diff
@@ -269,36 +246,20 @@ class RandomColorJitter(Augmentation):
...
-              and the original image 20% of the time.
-        """
-
-        if self.random_execute(0.8):
-            x = tf.image.random_brightness(x, 0.8)
-            x = tf.image.random_contrast(x, 0.4, 1.6)
-            x = tf.image.random_saturation(x, 0.4, 1.6)
-            x = tf.image.random_hue(x, 0.2)
+    def __call__(self, x):
+        if np.random.rand() < self.prob:
```
Comment (sachinprasadhs) [inline examples/vision/barlow_twins.py]: I don't think this is doing `RandomColorJitter`, few things from logic side are missing.
Instead try to use the layers avilable here https://keras.io/api/layers/preprocessing_layers/image_augmentation/random_color_jitter/

File: examples/vision/barlow_twins.py [hunk 2]
```diff
@@ -508,29 +442,71 @@ class BTDatasetCreator:
...
+                self.on_epoch_end()
 
-        return tf.data.Dataset.zip((a1, a2)).with_options(self.options)
+            def __len__(self):
+                return int(np.ceil(len(self.x) / self.batch_size))
 
+            def __getitem__(self, index):
+                batch = self.x[index * self.batch_size : (index + 1) * self.batch_size]
+                a1 = np.stack([self.augmentor(xi) for xi in batch])
+                a2 = np.stack([self.augmentor(xi) for xi in batch])
```
Comment (sachinprasadhs) [inline examples/vision/barlow_twins.py]: This is highly inefficient way to run.... looping over n times is not the ideal way to get the performance result.

**Review by sachinprasadhs — COMMENTED** [review]
There are few critical comments from Gemini, you need to address those.
Those comments are valid.
I added any missed comment from Gemini.
Once you address it, I will review again.

File: examples/vision/barlow_twins.py (hunk 2 above)
Comment (maitry63) [reply examples/vision/barlow_twins.py]: corrected

File: examples/vision/barlow_twins.py (hunk 1 above)
Comment (maitry63) [reply examples/vision/barlow_twins.py]: corrected

File: examples/vision/barlow_twins.py [hunk 3]
```diff
@@ -892,84 +846,90 @@ def build_twin() -> keras.Model:
...
+    z1 = encoder(input1)
+    z2 = encoder(input2)
 
-    def train_step(self, batch: tf.Tensor) -> tf.Tensor:
-        """train_step method.
+    z = layers.Concatenate(axis=1)([z1, z2])
 
-        Do one train step. Make model predictions, find loss, pass loss to
-        optimizer, and make optimizer apply gradients.
+    return keras.Model([input1, input2], z)
```
Comment (kharshith-k) [inline examples/vision/barlow_twins.py]: As per the tutorial's own architecture section (line 694–701), Barlow Twins is supposed to be encoder + projector MLP (3× dense-batchnorm-relu, 5000-dim). build_projector implements that MLP but is dead code — build_barlow_model feeds the raw ResNet34 encoder output straight into BarlowLoss. This silently changes the model architecture from what's documented and will materially hurt the accuracy this tutorial claims (62–64%), since the cross-correlation loss is designed to operate on high-dimensional projector embeddings, not raw conv features. Please wire build_projector (or equivalent) into build_barlow_model

Comment (maitry63) [reply examples/vision/barlow_twins.py]: @kharshith-k, thank you for your review! Fixed the implementation as per the tutorial architecture.
